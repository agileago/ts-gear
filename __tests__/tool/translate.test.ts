// import { cloneDeep } from 'lodash'

// import pontSchema from 'example/fixture/pontFixture.json'
import {
  translate,
  // gatherNonEnglishWords,
  // translateEngines,
  // generateTranslationMap,
  // updateSchema,
} from 'src/tool/translate'

describe.skip('translate by engines', () => {
  it('translate by baidu', async () => {
    expect(await translate('输出结果«查询参数»', 'baidu')).toBe('Output result « query parameter »')
  }, 5000)

  it('translate by google', async () => {
    expect(await translate('输出结果«查询参数»', 'google')).toBe('Output result «Query parameters»')
  }, 5000)
})

// it('gather cn words', () => {
//   const schema = cloneDeep(pontSchema) as JSONSchema
//   expect(gatherNonEnglishWords(schema)).toEqual([
//     '通用请求参数«查询参数»',
//     '输出参数vo',
//     'abc输出参数',
//     ' 中英文 混合 带 空格 Vo ',
//     '查询参数',
//     '通用请求参数token«输出参数vo»',
//     'DataTransOutput«输出参数vo»',
//   ])
// })

// describe('mock baidu translate', () => {
//   let originBaiduEngine = translateEngines.baidu
//   beforeAll(() => {
//     originBaiduEngine = translateEngines.baidu
//     // mock translate
//     ;(translateEngines as any).baidu = {
//       translate: () => {
//         return Promise.resolve({ result: ['system development'] })
//       },
//     }
//   })
//   afterAll(() => {
//     translateEngines.baidu = originBaiduEngine
//   })

//   it('generate translate map', async () => {
//     const words = ['系统开发']
//     const map = await generateTranslationMap(words, 'baidu')
//     expect(map).toEqual({
//       系统开发: 'system development',
//     })
//   })

//   it('update schema', async () => {
//     const schema = cloneDeep(pontSchema) as JSONSchema
//     // console.log(Object.keys(schema.definitions!))
//     const wordsMap = {
//       '通用请求参数«查询参数»': 'general request«query parameter»',
//       输出参数vo: 'output parameter vo',
//       abc输出参数: 'abc output parameter',
//       ' 中英文 混合 带 空格 Vo ': 'chinese english mixin space Vo',
//       查询参数: 'query parameter',
//       '通用请求参数token«输出参数vo»': 'general request «TokenVo»',
//       'DataTransOutput«输出参数vo»': 'DataTransOutput «OutputParameterVo»',
//     }
//     updateSchema(schema, wordsMap)
//     // console.log(schema)
//     expect(schema).toMatchSnapshot()
//   })
// })
