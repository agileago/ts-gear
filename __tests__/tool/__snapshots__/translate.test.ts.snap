// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock baidu translate update schema 1`] = `
Object {
  "basePath": "/",
  "definitions": Object {
    "DataTransOutput": Object {
      "description": "带数据的返回数据",
      "properties": Object {
        "data": Object {
          "description": "返回数据",
          "type": "object",
        },
        "transCode": Object {
          "description": "错误码。
100000 成功
200000 入参不合法
400000 权限不足
500000 服务失败",
          "format": "int32",
          "type": "integer",
        },
        "transMessage": Object {
          "description": "错误信息。成功：“成功” 失败：“失败对应的msg”",
          "type": "string",
        },
        "transMessageDetail": Object {
          "description": "信息详情”",
          "type": "string",
        },
      },
      "required": Array [
        "transCode",
        "transMessage",
        "transMessageDetail",
      ],
      "type": "object",
    },
    "DataTransOutput «OutputParameterVo»": Object {
      "description": "带数据的返回数据",
      "properties": Object {
        "data": Object {
          "$ref": "#/definitions/output parameter vo",
          "description": "返回数据",
        },
        "transCode": Object {
          "description": "错误码。
100000 成功
200000 入参不合法
400000 权限不足
500000 服务失败",
          "format": "int32",
          "type": "integer",
        },
        "transMessage": Object {
          "description": "错误信息。成功：“成功” 失败：“失败对应的msg”",
          "type": "string",
        },
        "transMessageDetail": Object {
          "description": "信息详情”",
          "type": "string",
        },
      },
      "required": Array [
        "transCode",
        "transMessage",
        "transMessageDetail",
      ],
      "type": "object",
    },
    "ResultListMap": Object {
      "properties": Object {
        "description": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "Result«List«Map»»": Object {
      "properties": Object {
        "description": Object {
          "type": "string",
        },
        "result": Object {
          "items": Object {
            "$ref": "#/definitions/Map",
          },
          "type": "array",
        },
        "returnCode": Object {
          "format": "int32",
          "type": "integer",
        },
      },
      "type": "object",
    },
    "abc output parameter": Object {
      "properties": Object {
        "dueDay": Object {
          "description": "开卡日",
          "type": "string",
        },
        "requestChannel": Object {
          "description": "进件渠道",
          "type": "string",
        },
      },
      "required": Array [
        "dueDay",
        "requestChannel",
      ],
      "type": "object",
    },
    "chinese english mixin space Vo": Object {
      "properties": Object {
        "dueDay": Object {
          "description": "开卡日",
          "type": "string",
        },
        "requestChannel": Object {
          "description": "进件渠道",
          "type": "string",
        },
      },
      "required": Array [
        "dueDay",
        "requestChannel",
      ],
      "type": "object",
    },
    "general request «TokenVo»": Object {
      "properties": Object {
        "bizParamVo": Object {
          "$ref": "#/definitions/output parameter vo",
        },
      },
      "type": "object",
    },
    "output parameter vo": Object {
      "properties": Object {
        "dueDay": Object {
          "description": "开卡日",
          "type": "string",
        },
        "requestChannel": Object {
          "description": "进件渠道",
          "type": "string",
        },
      },
      "required": Array [
        "dueDay",
        "requestChannel",
      ],
      "type": "object",
    },
    "query parameter": Object {
      "properties": Object {
        "data": Object {
          "additionalProperties": Object {
            "$ref": "#/definitions/DataTransOutput",
            "type": "array",
          },
          "type": "object",
        },
        "dueDay": Object {
          "description": "开卡日",
          "type": "string",
        },
        "requestChannel": Object {
          "description": "进件渠道",
          "type": "string",
        },
      },
      "required": Array [
        "dueDay",
        "requestChannel",
      ],
      "type": "object",
    },
  },
  "host": "http://localhost",
  "info": Object {
    "contact": Object {
      "email": "67890211@qq.com",
      "name": "管理平台",
      "url": "http://localhost",
    },
    "description": "",
    "license": Object {
      "name": "license",
      "url": "http://localhost",
    },
    "termsOfService": "http://localhost",
    "title": "接口文档",
    "version": "1.0",
  },
  "paths": Object {
    "/api/core/asset/credit/query/pastCreditCardBillGather": Object {
      "post": Object {
        "consumes": Array [
          "application/json;charset=UTF-8",
        ],
        "description": "往期账单汇总",
        "operationId": "pastCreditCardBillGatherUsingPOST",
        "parameters": Array [
          Object {
            "description": "requestParamVo",
            "in": "body",
            "name": "body",
            "required": true,
            "schema": Object {
              "$ref": "#/definitions/general request«query parameter»",
            },
          },
        ],
        "produces": Array [
          "application/json;charset=UTF-8",
        ],
        "responses": Object {
          "200": Object {
            "description": "OK",
            "schema": Object {
              "$ref": "#/definitions/DataTransOutput",
            },
          },
          "201": Object {
            "description": "Created",
          },
          "401": Object {
            "description": "Unauthorized",
          },
          "403": Object {
            "description": "Forbidden",
          },
          "404": Object {
            "description": "Not Found",
          },
        },
        "summary": "往期账单汇总",
        "tags": Array [
          "credit-query-controller",
        ],
      },
    },
    "/zzjy/all-msg/getAllMsgForMap": Object {
      "post": Object {
        "consumes": Array [
          "application/json",
        ],
        "operationId": "getAllMsgForMapUsingPOST",
        "parameters": Array [
          Object {
            "description": "accountTime",
            "in": "query",
            "name": "accountTime",
            "required": true,
            "type": "string",
          },
          Object {
            "description": "type",
            "in": "query",
            "name": "type",
            "required": true,
            "type": "string",
          },
        ],
        "produces": Array [
          "*/*",
        ],
        "responses": Object {
          "200": Object {
            "description": "OK",
            "schema": Object {
              "$ref": "#/definitions/Result«List«Map»»",
            },
          },
          "201": Object {
            "description": "Created",
          },
          "401": Object {
            "description": "Unauthorized",
          },
          "403": Object {
            "description": "Forbidden",
          },
          "404": Object {
            "description": "Not Found",
          },
        },
        "summary": "站址经营-汇总首页",
        "tags": Array [
          "credit-query-controller",
        ],
      },
    },
  },
  "swagger": "2.0",
  "tags": Array [
    Object {
      "description": "Credit Query Controller",
      "name": "credit-query-controller",
    },
  ],
}
`;
